### ISSUE TITLE
Password-protect the AdGuard settings

### ISSUE THREAD

**Issue by dnovitskyi** [AdguardTeam/AdguardForAndroid#1703]
This is a feature request sent by one of the users:

" Hey there! maybe it would make sense to put in the option to set a
password for the DNS settings (in the Android app), so the "adguard
family dns" can't be turned off so easily. "

**Comment by Revertron** [comment]
May be it would be better to make all AdGuard password protected then? :)

**Comment by TPS** [comment]
Perhaps integrate w/ fingerprint/face lock, for such devices?

**Comment by ryan222chen** [comment]
******I hope the app can add a feature to set a login password like the PC version. Thank you.

**Comment by kozv** [comment]
I am a porn addict. Please implement this feature.
This feature will help me.

When I slip my addiction, I disable the protection myself. If parental control locks the settings of adguard, I will not be able to change the settings and open porn sites.

I will ask friends and family who I told about my porn addiction to manage the passwords. I will have to contact them to turn it off. I can't turn it off right away. The time when I'm about to slip will end in about 30 minutes. This way I won't slip.

If Adguard for Android implements Parental Control, it will be a big help to me and many people in the Nofap Community who have similar problems to me. Thank you.

**Comment by aalso132** [comment]
8 years and this feature still hasn't been implemented? 
It would really help if AdGuard took this issue seriously. Currently, the app can be bypassed by various means (including quick settings, on android) and password-protecting the app would allow it to be used as a parental control app. It's a real shame and incomprehensible that this subject isn't a priority (when many other applications manage to implement this type of functionality very well).
Please listen to us!

Translated with DeepL.com (free version)

**Comment by TPS** [comment]
>  allow it to be used as a parental control app

To do that, it _@ least_ needs to be combined w/ [device](https://github.com/AdguardTeam/AdguardForAndroid/issues/5285) [admin](https://github.com/AdguardTeam/AdguardForAndroid/issues/5320) so that it can't be uninstalled, & further so clearing app data or stopping app outright wouldn't disable parental control setup inside anyway.  Frankly, it would need to be a separate app, architected for child protection _by default_ such that [said app _must_ be enabled for data to function, which has not been implemented in AG](https://github.com/AdguardTeam/AdguardForAndroid/issues/520).

**Comment by scorpio06** [comment]
If there's an update, it only takes my kids 5 seconds to turn off the app to have full access without anything leaking.

Really, all security goes down the drain if they turn it off so easily and quickly that a 5-year-old can do it in 1, 2, or 3 seconds.

hay alguna actualización, mis hijos no les toma más de 5 segundos apagar la aplicación para tener acceso total sin nada que lo filtre.

realmente toda la seguridad se va al caño si la apagan tan fácil y rápido que un niño de 5 años lo hace en 1, 2 por 3.

**Comment by TPS** [comment]
Unfortunately, some security features are a _byproduct_ of AG's featured goal of blocking ads.  I suspect you expect too much.

**Comment by AZIZxxp** [comment]
Yes, this is a feature that is required to set a password so that children cannot stop parental control. Thank you.

**Comment by HAYWIRE2466** [comment]
I do like this feature. I already have an app that runs as a device admin. What else is another one? If available, those who do not want it can disable it.

**Comment by gitfirewall** [comment]
please implement this feature

**Comment by lostcowboy** [comment]
Issue #5556 was recently closed because someone thought it was the same as this issue. I disagree. This issue #1703 is about setting a password to change DNS settings. #5556 was about kids using the AdGuard apps built in browser to get around Google  Parental Controls. It was suggested that the browser should be password-protected. That is the only thing similar about them.

**Comment by TPS** [comment]
#5556's definitively answered in https://github.com/AdguardTeam/AdguardForAndroid/issues/2740#issuecomment-1046255722

This issue, as you say, has no relation to that.
